web/packages/teleport/src/services/api/api.ts

```diff
@@ -111,6 +111,7 @@ const api = {
       ...params,
       headers: {
         ...params.headers,
+        ...defaultRequestOptions.headers,
```

Comment:
As I understand this code, we make a request, it fails due to lack of a webauthn assertion, so we get it and then have to repeat the request with MFA header added.

Unlike in other cases, we want the MFA header to be merged with whatever headers were used. But as we discussed in the other thread, `fetch` overrides the headers instead of merging them. Hence the default headers being used here.

But this code makes it impossible to make a request _without default headers_ to an endpoint gated by MFA. I don't know if it's a big deal, but we could easily circumvent that. Instead of stuffing the MFA header here, we could pass it as a new, third option to `fetch` which just adds the MFA header after auth headers.